### PR TITLE
[FIX] account: Accounting entry with Analytical tag is not distributed

### PR DESCRIPTION
**Pull request by nboulif**
Issue

	- Install "Accounting" module
	- Activate in settings "Analytic Accounting" and "Analytic Tags"
	- Create an analytical tag with distribution rules
	- Create a Bill or Invoice
	- Add any product
	- Add the analytic tag created above to the move line, then post
	- Go to Accounting -> Reporting -> Profir and Loss
	- Add one of the account related to "analytic tag distribution" as "Analytic" filter

	Missing entries; balance not updated.

Cause

	When the 'Analytic tag' have an 'Analytic distribution',
	the "Analytic Accounts" are related to the tag and not
	move line.

Solution

	When "Analytics accounts" are provided, fetch first if move line
	have the analytic account, else, if analytic tag is set on move line,
	fetch analytic accounts related to this tag.

opw-2324141

### CODE REVIEW

**Comment by smetl** [comment]
@qdp-odoo I let you check this because I really don't know what is correct or not^^

**Review by nim-odoo — COMMENTED** [review]
Hum, I have the feeling nobody really used this feature up to now :smile: It's the first time I see it. Maybe @jorenvo can enlighten us since you first committed it?

**Comment by qdp-odoo** [comment]
functionally speaking, I confirm there's something strange in here... the patch needs to be thoroughly tested though

**Comment by jorenvo** [comment]
I don't think this is correct. The analytic distribution is meant to auto-generate multiple analytic lines based on a percentage. You set it on a tag. E.g.:

- Tag 1
  - 60% analytic account 1
  - 40% analytic account 2

Adding Tag 1 to an `account.invoice.line` of $100 results in two `account.analytic.line`s:

- $60 in analytic account 1
- $40 in analytic account 2

This is working correctly and you can see this in the Analytic Report.

With this PR it will show up in the P&L as the full $100 in the analytic account you filter on. That's not correct.

If you want this to work then you have to calculate how much of the line is in the filtered analytic accounts. It's possible in theory because there's a [`move_id` on `account.analytic.line`](https://github.com/odoo/odoo/blame/ebbd74a38a23cccea7604522bbffa473f6418cf5/addons/account/models/account_move.py#L1411). There's no opposite o2m so you have to manually handle that in the financial reports. By knowing the analytic lines and their analytic accounts you can calculate the amount based on the set filters.

However, in that case you end up with strange situations. These distributions purposefully don't have to add up to 100%. I can have a distribution like:

- Tag 2
  - 90% analytic account 1
  - 100% analytic account 2

It's ok for analytic accounting. But if you now open a financial report like the P&L and filter on both analytic account 1 and 2 you'd display 190% of the amount which is probably a bad thing. 

So in summary I'm not sure how to solve it or if it should be solved in the first place. Financial reports are based on `account.move.line` and there's no one to one correspondence to `account.analytic.line`. 

PS. this part was actually done by TBE, but I'm pretty sure I understand the feature. @tbe-odoo in case I said something terribly wrong feel free to jump in.

**Comment by qdp-odoo** [comment]
hoooooooooo

yes @jorenvo is totally right

the P&L filtered with an analytic account filters the journal items having that account (there's no journal items with that account if you only set a distribution tag, although you can put both)
the P&L filtered with an analytic tag shows the journal items having that distribution tag

if you want to have a reporting showing the percentage of journal items as per done in the distribution tag, you may want to use the `reporting \ management \ analytic report` instead... or the analytic items list view.

So in conclusion, the system works as intended and despite not giving clear information on what that does, there's no bug there. I'll be closing the bug report, as well as creating an internal task to improve the usability and make that clearer.

Thanks for your udnerstanding

**Comment by nboulif** [comment]
Alright :+1: Thank you all :wink:

**Comment by tbe-odoo** [comment]
You are right @jorenvo , thanks for your explanation :)
